### DISCUSSION TITLE
How to create admin user ?

### DISCUSSION THREAD

**Discussion by jm61** [pocketbase/pocketbase#575] [Q&A] [answered]
Hello to all, sorry for dumb question, I'm pretty get used to REST Api, I read the docs but I still can't understand how to create an admin user with Js sdk, everytime I see my Authorization in headers with 'User <token>'... thanks for help

**Comment by ganigeorgiev** [comment, marked as answer]
Admin accounts can be created either from the Admin UI or via the REST API (and SDK) **if you've previously authorized as admin**.

Here is an example with the JavaScript SDK:

```js
const client = new PocketBase("http://127.0.0.1:8090");

async function createAdmin() {
    try {
        // authenticate
        await client.admins.authViaEmail("admin@example.com", "1234567890");

        // create new admin
        const newAdmin = await client.admins.create({
            email:           "new_admin@example.com",
            password:        "1234567890",
            passwordConfirm: "1234567890",
            avatar:          5, // optional used only for visual distinction
        });
    } catch (err) {
        // do something with the error...
        console.log(err);
    }
}

createAdmin();
```

All Admin APIs could be found in https://pocketbase.io/docs/api-admins/. _The documentation will be updated in the near future with more examples and better structure._

**Reply by jm61** [reply]
yes I applied this code but still get "401 The request requires admin authorization token to be set."
at which part, I must apply the 'Authorization: Admin <token>' ? I'm logged on as user on PB... :-/

**Reply by ganigeorgiev** [reply]
You cannot create an admin with a regular user. that's why `await client.admins.authViaEmail("admin@example.com", "1234567890")` is necessary. 

Once you are authenticated as admin, your `client.authStore` will be updated automatically and will add the `Authorization: Admin ...` header for any following requests.

**Reply by jm61** [reply]
okay got it! thanks for your reactivity

**Reply by jm61** [reply]
"your client.authStore will be updated automatically and will add the Authorization: Admin ... header for any following requests."
It's the point I missed initially ;-)
